x-pack/plugins/apm/public/components/app/correlations/failed_transactions_correlations.tsx

```diff
@@ -456,54 +425,55 @@ export function FailedTransactionsCorrelations({
       <EuiSpacer size="s" />
 
       <CorrelationsProgressControls
-        progress={progress}
-        isRunning={isRunning}
-        onRefresh={startFetchHandler}
+        progress={progressNormalized}
+        isRunning={progress.isRunning}
+        onRefresh={startFetch}
         onCancel={cancelFetch}
       />
 
-      {ccsWarning && (
+      {response.ccsWarning && (
         <>
           <EuiSpacer size="m" />
+          {/* Failed transactions correlations uses ES aggs that are available since 7.15 */}
           <CrossClusterSearchCompatibilityWarning version="7.15" />
         </>
       )}
 
-      {inspectEnabled &&
-      selectedTerm?.pValue != null &&
-      (isRunning || correlationTerms.length > 0) ? (
-        <>
-          <EuiSpacer size="m" />
-          <Summary
-            items={[
-              <EuiBadge color="hollow">
-                {`${selectedTerm.fieldName}: ${selectedTerm.key}`}
-              </EuiBadge>,
-              <>{`p-value: ${selectedTerm.pValue.toPrecision(3)}`}</>,
-            ]}
-          />
-        </>
-      ) : null}
+      {showSummaryBadge &&
+        selectedTerm !== undefined &&
+        selectedTerm.pValue !== null && (
+          <>
+            <EuiSpacer size="m" />
+            <Summary
+              items={[
+                <EuiBadge color="hollow">
+                  {`${selectedTerm.fieldName}: ${selectedTerm.fieldValue}`}
+                </EuiBadge>,
+                <>{`p-value: ${selectedTerm.pValue.toPrecision(3)}`}</>,
+              ]}
+            />
+          </>
+        )}
 
       <EuiSpacer size="m" />
 
       <div data-test-subj="apmCorrelationsTable">
-        {(isRunning || correlationTerms.length > 0) && (
-          <CorrelationsTable<FailedTransactionsCorrelationValue>
+        {showCorrelationsTable && (
+          <CorrelationsTable<FailedTransactionsCorrelation>
             columns={failedTransactionsCorrelationsColumns}
             significantTerms={correlationTerms}
-            status={isRunning ? FETCH_STATUS.LOADING : FETCH_STATUS.SUCCESS}
+            status={
+              progress.isRunning ? FETCH_STATUS.LOADING : FETCH_STATUS.SUCCESS
```

Comment:
I'm wondering if we should just apply this status server side?

Reply:
`isRunning` is part of Kibana's search strategy contract, we cannot replace it, we could just add our own loading logic in addition on top of Kibana's customizable `rawResponse` which I'd like to avoid.